polaris-core/src/main/java/org/apache/polaris/core/storage/s3compatible/S3CompatibleCredentialsStorageIntegration.java

```diff
@@ -0,0 +1,164 @@
+import org.apache.polaris.core.storage.s3compatible.S3CompatibleStorageConfigurationInfo.CredsCatalogAndClientStrategyEnum;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.services.sts.StsClientBuilder;
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest;
+import software.amazon.awssdk.services.sts.model.AssumeRoleResponse;
+
+/** Credential vendor that supports generating */
+public class S3CompatibleCredentialsStorageIntegration
+    extends InMemoryStorageIntegration<S3CompatibleStorageConfigurationInfo> {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(S3CompatibleCredentialsStorageIntegration.class);
+  private StsClient stsClient;
+
+  public S3CompatibleCredentialsStorageIntegration() {
+    super(S3CompatibleCredentialsStorageIntegration.class.getName());
+  }
+
+  public void createStsClient(S3CompatibleStorageConfigurationInfo s3storageConfig) {
+
+    LOGGER.debug("S3Compatible - createStsClient()");
+    StsClientBuilder stsBuilder = software.amazon.awssdk.services.sts.StsClient.builder();
+    stsBuilder.region(
+        Region
+            .US_WEST_1); // default region to avoid bug, because most (all?) S3 compatible softwares
+    // do not care about regions
+    stsBuilder.endpointOverride(
+        URI.create(
+            s3storageConfig
+                .getS3Endpoint())); // S3Compatible - AWS STS endpoint is unique and different from
+    // the S3 Endpoint
+    stsBuilder.credentialsProvider(
+        StaticCredentialsProvider.create(
+            AwsBasicCredentials.create(
+                s3storageConfig.getS3CredentialsCatalogAccessKeyId(),
+                s3storageConfig.getS3CredentialsCatalogSecretAccessKey())));
+    this.stsClient = stsBuilder.build();
+    LOGGER.debug("S3Compatible - stsClient successfully built");
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public EnumMap<PolarisCredentialProperty, String> getSubscopedCreds(
+      @NotNull PolarisDiagnostics diagnostics,
+      @NotNull S3CompatibleStorageConfigurationInfo storageConfig,
+      boolean allowListOperation,
+      @NotNull Set<String> allowedReadLocations,
+      @NotNull Set<String> allowedWriteLocations) {
+
+    LOGGER.debug("S3Compatible - getSubscopedCreds - applying credential strategy");
+    EnumMap<PolarisCredentialProperty, String> propertiesMap =
+        new EnumMap<>(PolarisCredentialProperty.class);
+    propertiesMap.put(PolarisCredentialProperty.AWS_ENDPOINT, storageConfig.getS3Endpoint());
+    propertiesMap.put(
+        PolarisCredentialProperty.AWS_PATH_STYLE_ACCESS,
+        storageConfig.getS3PathStyleAccess().toString());
```

Comment:
These are catalog properties, not credential-vending properties. These should be set at the catalog-level when it is created. Those properties would then be passed into the FileIO when it is constructed.

Reply:
To be refactor for satisfying change to requested : boolean "SKIP_CREDENTIAL_SUBSCOPING_INDIRECTION".
I will try to move it in the catalog properties.

Reply:
failed to use catalog porperties, they are not forwarded at this class